src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,57 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int INITIAL_SIZE = 10;
+    private int currentSize = INITIAL_SIZE;
+    private int size = 0;
+    private T[] table = (T[]) new Object[currentSize];
```

Comment:
Casting `Object[]` to `T[]` will generate an unchecked warning at runtime. It's safe to suppress this warning if you ensure that only `T` instances are put into the array. However, it's better to encapsulate the cast in a separate method and suppress the warning there to avoid polluting the rest of your code.

Reply:
ok, done